kelpie-f446/Core/UserDrivers/ThrusterDriver.c

```diff
@@ -23,16 +24,29 @@ ThrusterInfo_t thrusterLookup[8] =
 		{THRUSTER8, 		1000,		2000,		1500,		1250,		1750,			1/0.0381,				TIM_CHANNEL_4,			&htim1,			&(TIM1->CCR4)}
 };
 
-PUBLIC void SetThrusterPWM(ThrusterID_t thrusterID, pwm_t pwm)
+PUBLIC result_t SetThrusterPWM(ThrusterID_t thrusterID, pwm_t pwm)
 {
 
 	*(thrusterLookup[thrusterID].timerReg) = pwm * thrusterLookup[thrusterID].counterValuePerPWM;
 	SerialPrintln("PWM: %d", *(thrusterLookup[thrusterID].timerReg));
-	HAL_TIM_PWM_Start(thrusterLookup[thrusterID].timerHandler, thrusterLookup[thrusterID].timerChannel);
+	if(HAL_TIM_PWM_Start(thrusterLookup[thrusterID].timerHandler, thrusterLookup[thrusterID].timerChannel) == HAL_OK){
+		return RESULT_OK;
+	}
+	return RESULT_ERR;
 }
 
 // Motor initialization sequence
-PUBLIC void ThrusterDriverInit()
+PUBLIC result_t ThrusterDriverInit()
 {
+	uint8_t numThrusters = NUM_THRUSTERS;
+	result_t ret = RESULT_OK;
 
+	while(numThrusters > 0){
+		numThrusters--;
+		DA_SetThrusterValue(numThrusters, THRUSTER_INIT_VALUE);
```

Comment:
thruster init value can be placed in the driver layer since it is hardware specific.